src/WebJobs.Script.WebHost/Middleware/NonAppServiceWorkerMiddleware.cs

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Azure.WebJobs.Script.Extensions;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Middleware
+{
+    public class NonAppServiceWorkerMiddleware
+    {
+        private const string DisguisedHostHeader = "DISGUISED-HOST";
+        private const string HostHeader = "HOST";
+        private const string ForwardedProtocolHeader = "X-Forwarded-Proto";
+        private readonly RequestDelegate _next;
+
+        public NonAppServiceWorkerMiddleware(RequestDelegate next)
+        {
+            _next = next;
+        }
+
+        public async Task Invoke(HttpContext httpContext)
+        {
+            if (httpContext.Request.Headers.ContainsKey(DisguisedHostHeader))
```

Comment:
Would be fantastic if you added a test for this. Either an end to end test if possible, and or a middleware unit test.